README.md

```diff
@@ -1,2 +1,73 @@
+
+### Module boundaries and scope
+When leveraging the JS module graph either as `import ...` or as `<scropt type="module">` we are provided a new scope in which the code available therein is separate from teh rest of the JS scope. Would it make sense to allow for a custom element registered via either via import mechanism or via attribute rules to have such a boundary? In this way we could allow for techniques like scoped registries to be available as if for free with the new scope being create at these boundaries by default...
+
+### Inline modules (CSS or JS or CEs) tagged as URL
+Declarative CSS module research has included inestigation into a sharing mechanism to [prevent multiple copies of the same CSS module](https://github.com/WICG/webcomponents/issues/939). The ability, and possible need, to do so would prevent duplication in a "bundled" context; whether that bundle was manual or built by tooling. ID references do not cross Shadow DOM boundaries, so a technique to cache and reference this content in those situations would be productive.
+
+## Conciderations
+
+### Polyfill/tooling to aid adoption
+Any solution put forward here would benefit greatly by being paired as soon as possible with a bundler plug-in or polyfill that would make it possible for consumers to leverage the technique _today_. Previous attempts at larger scale quality of life improvements in this area have suffered greatly from low x-browser investment that has made developer exploration, let along developer adoption, of advances difficult, if not impossible. In the case that this tool the form of a polyfill, it would be important that the polyfill were small enough to inline within the `<head>` of a document in order to mitigate any potention FOUC.
+
+### Is it possible to make components that are valuable to users with 0 JS?
+An oft confused (or similarly highly important) topic when dicussing "declarative" APIs is "does it work with no JS?" and that is something that will come up often in any conversations here. If we bind this approach to a `<script>` tag, does that imply that the "declarative" API requires JS? Could a browser vendor speak to whether or not `<script>` is the boundary at which JS being turned off the parser turns back, or is it possible that `asserttype` would allow `customelement` content within a `<script>` tag to still be run in that context? IF that is not possible, then this may already start to shape the aboslutely possibilities of this addition and point us towards alternate choices. The `<template>` tag is certainly an option for inline techniques but we would still be lacking a graph mechanism by which we could build this code. Long term web component developers might lament the loss of HTML Imports in this case, and it might be worth revisiting that concept where actual 0 JS a requirment.
```

Comment:
>If we bind this approach to a <script> tag,

needs a sample or at least link to. It is unclear how SCRIPT tag would interact with DCE.

Reply:
The reality that I can to in this conversation with myself if that a declarative custom element and a declarative web component _might_ be different things. Once you get not “declarative” and “web component” you are talking about a host of capabilities that may not be covered by declarative “custom element”. However, all cases seem to point to not having ANY `<script>` tags, which means you need to go all the way to declarative attributes, which puts a load of extra stress on a proposal here.